test/linear_samplers/proc_vec_row_uniform.jl

```diff
@@ -0,0 +1,29 @@
+# This file is part of RLinearAlgebra.jl
+
+module ProceduralTestLSVRUnifSampler
+
+using Test, RLinearAlgebra, Random
+
+Random.seed!(1010)
+
+@testset "LSVR Uniform Sampling -- Procedural" begin
+
+    # Verify appropriate super type
+    @test supertype(LinSysVecRowUnifSampler) == LinSysVecRowSampler
+
+    # Test construction
+    @test let
+        A = rand(10,3)
+        b = rand(10)
+        x = rand(3)
+
+        samp = LinSysVecRowUnifSampler()
+
+        α, β = RLinearAlgebra.sample(samp, A, b, x, 1)
+
+        true
```

Comment:
This would be great, and I think it is not clear to me how to do this. Suppose we seed it, then we are really only verifying the quantity for one particular value (which may change depending on the default random number generator). We could certainly do this though.

We would want to do this as a hypothesis test to see if the distribution of $\alpha$ and $\beta$ are what we think they are supposed to be, but I think we are not there just yet. 

This is why I call these tests "procedural" as they just ensure that the code is running rather than checking for the correctness of the output.

Reply:
@vp314 

I see. I was thinking more like recording the "coeff" vector output and then verify the output of the sample function is proper. In this case if we, for instance, change the implementation of sample, we make sure we have a sanity check if the behavior is modified. Otherwise the test will only fail if there is some error thrown upwards and, as a consequence, the test might not be that useful.

The issue then is what we discussed in the proposal, that Julia does not ensure that RNG has the same behavior over different architectures/versions of Julia, and it might happened that the test fails because the seed generates a different sequence (coeff) in a different machine/Julia version. In this case, perhaps is useful to implement some overriding to the RNG that allows us to generate a deterministic sequence to just verify the implementation is correct.

What are your thoughts?

Reply:
I agree, the current test is not very useful nor informative.

Suppose we use `sample` for this scheme as a wrapper for the RandBLAS function. Then using an RNG and seed would not help us in this situation. We could check the output types. But it is not clear to me that we should do this either (as this would depend on the input types and whether the outputs are explicitly returned). What do you think?

Eventually, we would want to (expensively) check the distribution of the resulting outputs using a statistical test (I think we are in agreement here).

Reply:
@dmaldona I think maybe we should leave this bad test as a place holder for now, until we are at a point where we can design it better.

Reply:
@vp314 this makes sense. I agree. We can circle back on it later.